server/hotrod/src/main/java/org/infinispan/server/hotrod/CacheDecodeContext.java

```diff
@@ -149,11 +149,14 @@ Response get() {
    Response getKeyMetadata() {
       CacheEntry<byte[], byte[]> ce = cache.getCacheEntry(key);
       if (ce != null) {
-         NumericVersion entryVersion = (NumericVersion) ce.getMetadata().version();
+         long version = 0;
+         EntryVersion entryVersion = ce.getMetadata().version();
+         if (entryVersion != null && entryVersion instanceof NumericVersion) {
```

Comment:
Yeah I agree, however it will always be 0 now :) at least there would be some sort of comparison possible. But I agree we should have another JIRA to look into this whole version part.